### PR TITLE
[SDK]Support eager flow serve environment variable connection

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If `pf flow serve` has EVC like this `--environment-variables AZURE_OPENAI_API_KEY='${open_ai_connection.api_key}'`.
According to different flow definition, there will be different behaviors:

- If connection in EVC exist in flow definition(in environment variables or node inputs), EVC will be resolved to actual value
- If not, EVC won't be resolved
This pull request primarily modifies the handling of metadata in the `promptflow` package, replacing the `entry` parameter with a more generic `data` dictionary in several methods. Additionally, new test cases have been added to check the functionality of environment variables in flow definitions.

Changes in handling of metadata:

* [`src/promptflow/promptflow/_core/tool_meta_generator.py`](diffhunk://#diff-5cd3c0319063421c9ff9210ca997e3f001e5d1e2c9f97b912f1a3f233f9ee5a6L495-R496): The `generate_flow_meta_dict_by_file` method now accepts a `data` dictionary instead of an `entry` string, and uses this to create the `flow_meta` dictionary. [[1]](diffhunk://#diff-5cd3c0319063421c9ff9210ca997e3f001e5d1e2c9f97b912f1a3f233f9ee5a6L495-R496) [[2]](diffhunk://#diff-5cd3c0319063421c9ff9210ca997e3f001e5d1e2c9f97b912f1a3f233f9ee5a6L506-R507)
* [`src/promptflow/promptflow/_sdk/entities/_eager_flow.py`](diffhunk://#diff-53d7ff49a8720f52ee715a6f63d5e389750e44c578a907076d435e541f0163eeL82): A TODO comment about supporting environment variables has been removed.
* [`src/promptflow/promptflow/batch/_python_executor_proxy.py`](diffhunk://#diff-8c42522c7186705de9a6651343af6006be02b7342f72467c2785f23f93ea669eL43-R43): The `entry` parameter in the `_generate_flow_json` method has been replaced with `data`.
* [`src/promptflow/promptflow/core/_flow.py`](diffhunk://#diff-a1b7af0df2fd1f9f1d07a45071795e8dc440a303c7847abe5033b9348909712bL319-R322): The `entry` parameter in the `load` method has been replaced with `data`.
* [`src/promptflow/promptflow/core/_utils.py`](diffhunk://#diff-c30ffbd98c271f76d167e48a57e3f65d936fb9ad3b75834cb1efa8a38307976eL23-R30): The `entry` parameter has been replaced with `data` in several methods, including `_generate_meta_from_file`, `_generate_flow_meta`, and `generate_flow_meta`. [[1]](diffhunk://#diff-c30ffbd98c271f76d167e48a57e3f65d936fb9ad3b75834cb1efa8a38307976eL23-R30) [[2]](diffhunk://#diff-c30ffbd98c271f76d167e48a57e3f65d936fb9ad3b75834cb1efa8a38307976eL40-R40) [[3]](diffhunk://#diff-c30ffbd98c271f76d167e48a57e3f65d936fb9ad3b75834cb1efa8a38307976eL62-R62) [[4]](diffhunk://#diff-c30ffbd98c271f76d167e48a57e3f65d936fb9ad3b75834cb1efa8a38307976eL81-R81) [[5]](diffhunk://#diff-c30ffbd98c271f76d167e48a57e3f65d936fb9ad3b75834cb1efa8a38307976eL92-R92) [[6]](diffhunk://#diff-c30ffbd98c271f76d167e48a57e3f65d936fb9ad3b75834cb1efa8a38307976eL102-R102)

Addition of new test cases:

* [`src/promptflow/tests/sdk_cli_test/conftest.py`](diffhunk://#diff-9dced3762e0027d9b6b131576702c5290a82d72c20b2a828c67aa682d31d8e40R244-R268): New fixtures have been added for testing environment variables in flow definitions.
* [`src/promptflow/tests/sdk_cli_test/e2etests/test_flow_serve.py`](diffhunk://#diff-8f175978a83627600049f94831e06a3c9e3307cbde190fe5fbb187571ac514baR634-R667): New test cases have been added to check the functionality of environment variables in flow definitions.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/sdk_cli_azure_test/conftest.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def create_serving_client_with_connections(model_name, mocker: MockerFixture, co
     with mock.patch(func) as mock_cred_func:
         mock_cred_func.return_value = None
         app = create_serving_app(
-            environment_variables={"API_TYPE": "${azure_open_ai_connection.api_type}"},
```
Comment (D-W-) [inline src/promptflow/tests/sdk_cli_azure_test/conftest.py:235]: @wxpjimmy please help to take a look at this

File: src/promptflow/promptflow/_sdk/_submitter/utils.py [hunk 2]
```diff
@@ -252,14 +252,13 @@ def init_env(cls, environment_variables):
...
+        client=None,
+        *,
+        connections_to_ignore=None,
+        connections_to_add: List[str] = None,
+        environment_variables_overrides: Dict[str, str] = None,
     ) -> dict:
-        from promptflow._sdk.entities._flow import FlexFlow
-
-        if isinstance(flow, FlexFlow):
-            # TODO(2898247): support prompt flow management connection for eager flow
```
Comment (elliotzh) [inline src/promptflow/promptflow/_sdk/_submitter/utils.py:260]: this is now supported?

File: src/promptflow/promptflow/core/_serving/flow_invoker.py [hunk 3]
```diff
@@ -75,14 +76,21 @@ def _init_connections(self, connection_provider):
             config = {"connection.provider": connection_provider} if connection_provider else None
             self.logger.info(f"Getting connections from pf client with provider from args: {connection_provider}...")
             connections_to_ignore = list(self.connections.keys())
+            self.logger.debug(f"Flow invoker connections: {self.connections.keys()}")
             connections_to_ignore.extend(self.connections_name_overrides.keys())
+            self.logger.debug(f"Flow invoker connections name overrides: {self.connections_name_overrides.keys()}")
+            self.logger.debug(f"Ignoring connections: {connections_to_ignore}")
```
Comment (wangchao1230) [inline src/promptflow/promptflow/core/_serving/flow_invoker.py:82]: what's this scenario?

File: src/promptflow/promptflow/contracts/flow.py [hunk 4]
```diff
@@ -546,15 +546,20 @@ def get_environment_variables_with_overrides(
                 environment_variables[k] = v
         return environment_variables
 
-    def get_connection_names(self):
-        """Return connection names."""
+    def get_connection_names(self, environment_variables_overrides: Dict[str, str] = None):
+        """Return connection names with environment variables overrides.
```
Comment (wangchao1230) [inline src/promptflow/promptflow/contracts/flow.py:550]: Return connection names used in this flow.

File: src/promptflow/promptflow/_sdk/_submitter/utils.py (hunk 2 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/_submitter/utils.py:260]: Yes, flex flow support prompt flow management connection by environment variable connection

File: src/promptflow/promptflow/contracts/flow.py [hunk 5]
```diff
@@ -546,15 +546,20 @@ def get_environment_variables_with_overrides(
                 environment_variables[k] = v
         return environment_variables
 
-    def get_connection_names(self):
-        """Return connection names."""
+    def get_connection_names(self, environment_variables_overrides: Dict[str, str] = None):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/contracts/flow.py:549]: add docstring on the parameter: environment_variables_overrides

File: src/promptflow/promptflow/core/_serving/flow_invoker.py (hunk 3 above)
Comment (D-W-) [reply src/promptflow/promptflow/core/_serving/flow_invoker.py:82]: Offline synced, used for connection object override in flow-as-a-func

**Review by thy09 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**